flocker/node/functional/test_docker.py

```diff
@@ -627,6 +627,7 @@ def run_registry(self):
         :return: A ``Registry`` describing the registry which was started.
         """
         registry_name = random_name(self)
+        external_port = find_free_port()[1]
```

Comment:
This introduces a different race condition: between the time we find a free port and when we start the container, something else might come along and use that port. The way to avoid this is to bind to port 0 and somehow discover the port, as the original code was doing.